tests/vmi_hook_sidecar_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+const hookSidecarImage = "example-hook-sidecar"
+
+var _ = Describe("HookSidecars", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	var vmi *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		vmi = tests.NewRandomVMIWithEphemeralDisk(tests.RegistryDiskFor(tests.RegistryDiskAlpine))
+		vmi.ObjectMeta.Annotations = map[string]string{
+			"hooks.kubevirt.io/hookSidecars": fmt.Sprintf("%s/%s:%s", tests.KubeVirtRepoPrefix, hookSidecarImage, tests.KubeVirtVersionTag),
```

Comment:
I think this test is failing because the string isn't created correctly.  Shouldn't it be something like this?
```
fmt.Sprintf("[{\"image\": \"%s/%s:%s\", \"imagePullPolicy\": \"IfNotPresent\" }]", tests.KubeVirtRepoPrefix, hookSidecarImage, tests.KubeVirtVersionTag)
```

Reply:
Oh my... Sorry